api/app/Models/Ads/AdBase.php

```diff
@@ -1,239 +1,314 @@
+                : null;
+            $this->adType = isset($row[$propertyColumns->adType])
+                ? htmlspecialchars($row[$propertyColumns->adType])
+                : null;
+            $this->condition = isset($row[$propertyColumns->condition])
+                ? htmlspecialchars($row[$propertyColumns->condition])
+                : null;;
+            $this->title = isset($row[$propertyColumns->title])
+                ? htmlspecialchars($row[$propertyColumns->title])
+                : null;
+            if (isset($row[$propertyColumns->description])) {
+                $this->description = str_replace("\n\r", "\n", $row[$propertyColumns->description]);
+                $this->description = str_replace("\n", "<br/>", $this->description);
+            } else {
+                $this->description = null;
             }
+            $this->price = isset($row[$propertyColumns->price])
+                ? htmlspecialchars($row[$propertyColumns->price])
+                : null;
+            $this->images = isset($row[$propertyColumns->imagesRaw])
+                ? explode(PHP_EOL, $row[$propertyColumns->imagesRaw])
+                : [];
+            $this->videoURL = isset($row[$propertyColumns->videoURL])
+                ? htmlspecialchars($row[$propertyColumns->videoURL])
+                : null;
+            
+            $this->address = $this->getFullAddress($row, $propertyColumns);
+            
+            $this->avitoId = isset($row[$propertyColumns->avitoManualID])
+                ? htmlspecialchars($row[$propertyColumns->avitoManualID])
+                : null;
+            $this->adStatus = isset($row[$propertyColumns->paidControl])
+                ? htmlspecialchars($row[$propertyColumns->paidControl])
+                : null;
+            
+            if (isset($row[$propertyColumns->placementType])) {
+                switch ($row[$propertyColumns->placementType]) {
+                    case "Пакет":
+                        $this->placementType = "Package";
+                        break;
+                    case "Пакет или кошелек":
+                        $this->placementType = "PackageSingle";
+                        break;
+                    case "Кошелек":
+                        $this->placementType = "Single";
+                        break;
+                    default:
+                        $this->placementType = null;
+                }
+            }
+            
+            $this->messages = isset($row[$propertyColumns->messages])
+                ? htmlspecialchars($row[$propertyColumns->messages])
+                : null;
         }
-        else
-        {
-            $this->dateBegin = null;
-        }
-
-        if(isset($row[$propertyColumns->dateEnd]))
+        
+        public abstract function toAvitoXml(): string;
+        
+        protected function getFullAddress($row, TableHeader $propertyColumns, int $number = null): string
```

Comment:
I do not understand the logic. First, you have protected/private properties, then public methods, then protected/private methods.